README.md

```diff
@@ -391,12 +393,16 @@ previous example using `rmm::device_vector`:
   rmm::device_vector<int> vec(16, rmm::mr::thrust_allocator<int>(rmm::cuda_stream_default, &mr));
   RMM_CUDA_TRY(cudaSetDevice(1));
   ...
-  // ERROR: ~vec runs with device 1 active, but needs device 0 to be active
+  // No need to switch back to device 0 before ~vec runs
 }
 ```
 
-A correct example adds a call to `cudaSetDevice(0)` on the line of the error comment before the dtor
-for `~vec` runs.
+> [!NOTE]
```

Comment:
I added this note because I was bitten by this issue.

Reply:
@harrism to respond to your confused emoji, despite these changes ensuring that `allocate` runs with the correct device active, as soon as the allocation is done the old device is restored. I wasn't thinking through that `.resize()` (or any call that needs to `uninitialized_fill` the vector) also needs to run with the correct device active (otherwise the kernel launch is to the wrong device).

e.g.
```c++
auto vec = rmm::device_vector<int>(10);
cudaSetDevice(other_device);
vec.resize(20); // <- invalid, because this launches a kernel to initialize the new entries
```

I can just remove it if it's more confusing to leave it in, I guess this is the natural behaviour: you need to use the data with the appropriate device active.